src/app/settings/action-triggers/components/edit-action-triggers/edit-action-triggers.component.ts

```diff
@@ -0,0 +1,174 @@
+  @Input() public actions: Tv2PartAction[]
+  public submitting: boolean = false
+  public submitted: boolean = false
+  public keyPress: boolean = false
+  public selectedAction: Tv2PartAction | null
+
+  constructor(
+    private readonly fb: FormBuilder,
+    private readonly actionTriggerService: ActionTriggerService
+  ) {}
+
+  public actionForm: UntypedFormGroup = this.fb.group({
+    actionId: ['', [Validators.required]],
+    data: this.fb.group({
+      keys: this.fb.array([]),
+      actionArguments: '',
+      label: [''],
+    }),
+  })
+
+  public ngOnChanges(changes: SimpleChanges): void {
+    const actionTriggerChange: SimpleChange | undefined = changes['selectedActionTrigger']
+    if (actionTriggerChange) {
+      const action: ActionTriggerWithActionInfo<KeyboardAndSelectionTriggerData> | null = actionTriggerChange.currentValue
+      this.clearFormArray(this.attributesArray)
+      if (action) {
+        this.patchValue(action)
+        this.selectedAction = action.actionInfo
+      } else {
+        this.actionForm.reset()
+        this.selectedAction = null
+      }
+    }
+  }
+
+  private patchValue(actionValue: ActionTrigger<KeyboardAndSelectionTriggerData>): void {
+    this.actionForm.patchValue(actionValue)
+    this.addArrValues(actionValue.data.keys, this.attributesArray)
+  }
+
+  private clearFormArray(formArray: UntypedFormArray): void {
+    while (formArray.length !== 0) {
+      formArray.removeAt(0)
+    }
+  }
+
+  public onKeyDown(event: KeyboardEvent): void {
+    event.preventDefault()
+    const newKeyCode: string = SHORTCUT_KEYS_MAPPINGS[event.code] ? SHORTCUT_KEYS_MAPPINGS[event.code] : event.code
+    if (this.keyPress) {
+      const currentKeys: string[] = this.actionForm?.get('data')?.get('keys')?.value ? this.actionForm?.get('data')?.get('keys')?.value : []
+      if (currentKeys.findIndex(keyCode => keyCode === newKeyCode) === -1) {
+        currentKeys.push(newKeyCode)
+        this.createKey(newKeyCode, this.attributesArray)
+      }
+    } else {
+      this.clearFormArray(this.attributesArray)
+      this.createKey(newKeyCode, this.attributesArray)
+      this.keyPress = true
+    }
+  }
+
+  private get attributesArray(): UntypedFormArray {
+    return this.actionForm?.get('data')?.get('keys') as UntypedFormArray
+  }
+
+  private addArrValues(items: string[], formArray: UntypedFormArray): void {
+    for (const item of items) {
+      this.createKey(item, formArray)
+    }
+  }
+
+  private createKey(item: string, formArray: UntypedFormArray): void {
+    const control = new UntypedFormControl(item)
+    formArray.push(control)
+  }
+
+  public selectNewActionTrigger(action: Tv2PartAction): void {
+    this.actionForm.patchValue({ actionId: action.id })
+    this.selectedAction = action
+  }
+
+  public onKeyUp(): void {
+    this.keyPress = false
+  }
+
+  public get isUpdateAction(): boolean {
```

Comment:
Suggestion to name: `isUpdateForm`

And question: When will the selected action trigger have been set, but not the action trigger id?

Reply:
I update the name. I keep it general, (isUpdateAction), because I reuse same method everywhere and it's easy to spot by my.
Answer: Never the value is only type ActionTrigger or undefined. I will remove check for the Id and only will check for existing of the ActionTrigger

Reply:
I should have given some reason behind the suggestion. When I read `isUpdateAction` I understand it as update describes a particular attribute of an action. `isUpdateForm` or `isActionUpdateForm` or `isUpdatingAction` indicates that they are describing the state of the form component.

Reply:
You're right. I have already changed it to `isUpdateForm`, and will keep that name.